test/commands/extended/is_promotable_test.py

```diff
@@ -0,0 +1,166 @@
+# coding=utf-8
+from __future__ import absolute_import, division, print_function, \
+  unicode_literals
+
+from unittest import TestCase
+
+import filters as f
+from filters.test import BaseFilterTestCase
+
+from iota import Iota, TransactionHash
+from iota.adapter import MockAdapter
+from iota.commands.extended.is_promotable import IsPromotableCommand
+from iota.filters import Trytes
+
+
+class IsPromotableRequestFilterTestCase(BaseFilterTestCase):
+  filter_type = IsPromotableCommand(MockAdapter()).get_request_filter
+  skip_value_check = True
+
+  def setUp(self):
+    super(IsPromotableRequestFilterTestCase, self).setUp()
+
+    # noinspection SpellCheckingInspection
+    self.transaction = (
+      'TESTVALUE9DONTUSEINPRODUCTION99999KPZOTR'
+      'VDB9GZDJGZSSDCBIX9QOK9PAV9RMDBGDXLDTIZTWQ'
+    )
+
+  def test_pass_happy_path(self):
+    """
+    Request is valid.
+    """
+    # Raw trytes are extracted to match the IRI's JSON protocol.
+    request = {
+      'tail': self.transaction,
+    }
+
+    filter_ = self._filter(request)
+
+    self.assertFilterPasses(filter_)
+    self.assertDictEqual(filter_.cleaned_data, request)
+
+  def test_pass_compatible_types(self):
+    """
+    Request contains values that can be converted to the expected
+    types.
+    """
+    filter_ = self._filter({
+      # Any TrytesCompatible value will work here.
+      'tail': TransactionHash(self.transaction),
+    })
+
+    self.assertFilterPasses(filter_)
+    self.assertDictEqual(
+      filter_.cleaned_data,
+
+      {
+        'tail': self.transaction,
+      },
+    )
+
+  def test_fail_empty(self):
+    """
+    Request is empty.
+    """
+    self.assertFilterErrors(
+      {},
+
+      {
+        'tail': [f.FilterMapper.CODE_MISSING_KEY],
+      },
+    )
+
+  def test_fail_unexpected_parameters(self):
+    """
+    Request contains unexpected parameters.
+    """
+    self.assertFilterErrors(
+      {
+        'tail': TransactionHash(self.transaction),
+
+        # SAY "WHAT" AGAIN!
+        'what': 'augh!',
```

Comment:
Aw, no fair!  You can't just copy-paste; you have to come up with your own joke here 😺